nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -274,26 +274,32 @@ in
     consoleMode = mkOption {
       default = "keep";
 
-      type = types.enum [
-        "0"
-        "1"
-        "2"
-        "5"
-        "auto"
-        "max"
-        "keep"
-      ];
+      type =
+        with types;
+        either (enum [
+          "0"
+          "1"
+          "2"
+          "auto"
+          "max"
+          "keep"
+        ]) ints.unsigned;
```

Comment:
My recommendation would be something along the lines of:

```nix
type = if versionAtLeast config.system.stateVersion "25.05" then
 types.enum [
        "0"
        "1"
        "2"
        "5"
        "auto"
        "max"
        "keep"
      ];
else
 with types;
        either (enum [
          "0"
          "1"
          "2"
          "auto"
          "max"
          "keep"
        ]) ints.unsigned;
```

which would then rigidly enforce the type signature on 25.05 and above. We could also introduce an assertion that checks `cfg` for values that are deprecated in 25.05, and throw an error upon evaluation.

It's worth bearing in mind #340597 here too.

Thoughts?